src/lib/blockchain/Inventory.ts

```diff
@@ -43,4 +45,21 @@ export class Inventory {
       {} as ItemSupply
     );
   }
+
+  public async getBalances(farmAddress: string) {
+    const batchAccounts = Array(IDS.length).fill(farmAddress);
+    const balances = await this.contract.methods.balanceOfBatch(batchAccounts, IDS).call();
+    const names = Object.keys(KNOWN_IDS) as InventoryItemName[];
+
+    return balances.reduce((items: Inventory, balance: string, index: number) => {
```

Comment:
Similar to above, I would suggest moving the game logic out of this Inventory class and into a separate helper class.

That helps make this file focussed on Blockchain interacts and also makes it easier to test the balances -> Inventory code.

E.g
```
const balances = await metamask.getInventory().getBalances(farmAddress);
const inventory = balancesToInventory(balances) // where the code would live
```

Reply:
created separate file `visitUtils.ts` and put `balancesToInventory()` there